### DISCUSSION TITLE
useForm for edit/create row modals

### DISCUSSION THREAD

**Discussion by ctretyak** [KevinVandy/mantine-react-table#112] [General]
Why you don't use https://mantine.dev/form/use-form for validation and submitting forms? It would be more convenient to validate form with zod or other schema based validator https://mantine.dev/form/validation/ . 

Also:
- we are not able to know if form is touched to show errors only in touched/submitted state
- form doesn't have button with type=submit and cannot submit by pressing "enter" key

**Comment by KevinVandy** [comment]
You can use it. I use it in my own table edit modals. It is just not installed by default. Render your own form in the renderEditModalContent override

**Reply by ctretyak** [reply]
Sure, but it looks logical to use mantine's useForm for forms in that kind of library. I'm trying to realise if there are any reasons why you didn't use it

**Comment by BahaaZidan** [comment]
I'd love for that to be the default. And I don't see any reason why it isn't. We can render our own form when it's a modal, but inline row editing is pretty much a nightmare. And I found myself having to override a lot of things to get it to work with `<form />` and `zod` (still trying :|)
